### PR TITLE
28 detailed map rp

### PR DESCRIPTION
**Pull request by yousef337**

### CODE REVIEW

**Comment by zaidalyafeai** [comment]
Looks good, one slight issue. We need a way to fix a selection i.e when I choose a country it shouldn't change to another country (hovering). The current setup makes it difficult to explore the table and avoid hovering over other countries.

**Comment by zaidalyafeai** [comment]
Looks good, how are we dealing with groups like gulf, levant, etc. ? Also, it might be useful to add a small text to indicate that you can click on the countries to fix a selection.

File: assets/js/utils/tableUtils.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+  
+  function getIcon(text){
+    const lower = text.toLowerCase()
+    if (icons[lower] != undefined || icons[lower] != "nan") {
+      return icons[lower];
+    } else {
+      return text;
+    }
+  }
```
Comment (zaidalyafeai) [inline assets/js/utils/tableUtils.js:41]: These functions seem to be repeated in many js files. Can we have a single file to export these from?

File: assets/js/amcharts.js [hunk 2]
```diff
@@ -1,107 +1,67 @@
...
-  );
-
-  worldSeries.mapPolygons.template.setAll({
-    fill: am5.color(0xaaaaaa),
-  });
-
-  worldSeries.events.on("datavalidated", () => {
-    chart.goHome();
-  });
+function transformDataToTableEntry(dataset) {
```
Comment (zaidalyafeai) [inline assets/js/amcharts.js:2]: Maybe transfer this function to `tableUtils.js`, doesn't seem to fit here.

File: _pages/graphs.html [hunk 3]
```diff
@@ -49,11 +53,22 @@ <h5 class="text-center">
             </select>
         </div>
 
+        <div id="selectionGroup" class="btn-group mt-3" role="group" aria-label="Basic radio toggle button group">
+            <input type="radio" class="btn-check" name="modeSelector" id="byCountry" autocomplete="off" checked>
+            <label class="btn btn-outline-primary" for="byCountry">By Country</label>
+          
+            <input type="radio" class="btn-check" name="modeSelector" id="byDialect" autocomplete="off">
+            <label class="btn btn-outline-primary" for="byDialect">By Dialect</label>
```
Comment (zaidalyafeai) [inline _pages/graphs.html:61]: I prefer `By Region`. Also, if we hover over the countries of a given region we should say the name of the region for example gulf.

File: assets/js/utils/mapDataUtil.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    pagingType: "numbers",
+    bInfo: false,
+    bDestroy: true,
+    createdRow: function (row, data, dataIndex) {
+      $("td:eq(9)", row).css("min-width", "200px");
+    },
+  });
+}
+
+function singleDialect(dialectedEntries) {
```
Comment (zaidalyafeai) [inline assets/js/utils/mapDataUtil.js:24]: Can we add some comments to show explain each function at high level?

File: _pages/graphs.html (hunk 3 above)
Comment (zaidalyafeai) [reply _pages/graphs.html:61]: Shouldn't Libya and Tunis be grouped with Morroco, Algeria, and Mauritania and labeled as NOR?

Comment (zaidalyafeai) [reply _pages/graphs.html:61]: Sure

Comment (yousef337) [reply _pages/graphs.html:61]: NOR is too large to be considered one region, especially with the variation of dialects, Moroccan and Libyan are quite different, at least from my perspective.

Comment (zaidalyafeai) [reply _pages/graphs.html:61]: I didnt come up with this, it is from literature.

**Comment by zaidalyafeai** [comment]
Fixed some issues related to missing labels. You can resolve the conflicts and merge to main.

**Comment by zaidalyafeai** [comment]
Going to close this for now.

**Pull request closed without merging**